### PR TITLE
gh-102444: Fix minor bugs in `test_typing` highlighted by pyflakes

### PR DESCRIPTION
**Pull request by AlexWaygood**
* Issue: gh-102444

### CODE REVIEW

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -7077,16 +7071,6 @@ class C:
...
-    def test_hash_eq(self):
-        self.assertEqual(len({Annotated[int, 4, 5], Annotated[int, 4, 5]}), 1)
-        self.assertNotEqual(Annotated[int, 4, 5], Annotated[int, 5, 4])
-        self.assertNotEqual(Annotated[int, 4, 5], Annotated[str, 4, 5])
-        self.assertNotEqual(Annotated[int, 4], Annotated[int, 4, 4])
-        self.assertEqual(
-            {Annotated[int, 4, 5], Annotated[int, 4, 5], Annotated[T, 4, 5]},
-            {Annotated[int, 4, 5], Annotated[T, 4, 5]}
-        )
-
```
Comment (AlexWaygood) [inline Lib/test/test_typing.py:7089]: This test is exactly identical to `test_hash_eq` on line 7015:

https://github.com/python/cpython/blob/7894bbe94ba319eb650f383cb5196424c77b2cfd/Lib/test/test_typing.py#L7015-L7023

File: Lib/test/test_typing.py [hunk 2]
```diff
@@ -473,7 +473,6 @@ def test_var_substitution(self):
 
     def test_bad_var_substitution(self):
         T = TypeVar('T')
-        P = ParamSpec("P")
```
Comment (AlexWaygood) [inline Lib/test/test_typing.py:476]: `P` here is currently an unused variable. `ParamSpec.__typing_subst__` is tested in another place in this file:

https://github.com/python/cpython/blob/7894bbe94ba319eb650f383cb5196424c77b2cfd/Lib/test/test_typing.py#L7528-L7539

File: Lib/test/test_typing.py [hunk 3]
```diff
@@ -7515,7 +7499,6 @@ class Y(Generic[P, T]):
         self.assertEqual(B.__args__, ((int, str,), Tuple[bytes, float]))
 
     def test_var_substitution(self):
-        T = TypeVar("T")
```
Comment (AlexWaygood) [inline Lib/test/test_typing.py:7518]: `T` here is currently an unused variable. `TypeVar.__typing_subst__` is tested more fully elsewhere:

https://github.com/python/cpython/blob/7894bbe94ba319eb650f383cb5196424c77b2cfd/Lib/test/test_typing.py#L461-L472

File: Lib/test/test_typing.py [hunk 4]
```diff
@@ -4918,7 +4913,6 @@ def test_overload_registry_repeated(self):
 # Definitions needed for features introduced in Python 3.6
 
 from test import ann_module, ann_module2, ann_module3, ann_module5, ann_module6
-import asyncio
```
Comment (kumaraditya303) [inline Lib/test/test_typing.py:4921]: Typing  in Python is definitely synchronous ;)

File: Lib/test/test_typing.py [hunk 5]
```diff
@@ -1307,7 +1304,7 @@ def test_callable_args_are_correct(self):
         i = Callable[[None], *Ts]
         j = Callable[[None], Unpack[Ts]]
         self.assertEqual(i.__args__, (type(None), *Ts))
-        self.assertEqual(i.__args__, (type(None), Unpack[Ts]))
+        self.assertEqual(j.__args__, (type(None), Unpack[Ts]))
```
Comment (AlexWaygood) [inline Lib/test/test_typing.py:1307]: The `j` variable is currently unused. Given the pattern of the assertions immediately below, it seems pretty clear that it was meant to be used here.

**Review by sobolevn — APPROVED** [review]
Thank you!

**Comment by AlexWaygood** [comment]
Thanks for the review @sobolevn!

**Pull request merged**